static/js/containers/HomePage_test.js

```diff
@@ -1,83 +1,191 @@
-
-    for (const sortType of VALID_POST_SORT_TYPES) {
-      await listenForActions(
-        [
-          actions.frontpage.get.requestType,
-          actions.frontpage.get.successType,
-          SET_POST_DATA
-        ],
-        () => {
-          const select = wrapper.find(".post-list-title").find("SortPicker")
-          select.props().updateSortParam(sortType, {
+  describe("sorting", () => {
+    it("uses hot sorting by default", async () => {
+      const { inner } = await render()
+      assert.equal(inner.find("PostSortPicker").props().value, POSTS_SORT_HOT)
+    })
+
+    it("uses the query parameter to set the sort", async () => {
+      const { inner } = await render(
+        {},
+        {
+          location: {
+            search: "?sort=some_sort"
+          }
+        }
+      )
+      assert.equal(inner.find("PostSortPicker").props().value, "some_sort")
+    })
+
+    it("should switch the sorting method when an option is selected", async () => {
+      const { inner } = await render()
+      for (const sortType of VALID_POST_SORT_TYPES) {
+        inner
+          .find("PostSortPicker")
+          .props()
+          .updateSortParam(sortType, {
             preventDefault: helper.sandbox.stub()
           })
+
+        assert.equal(helper.currentLocation.search, `?sort=${sortType}`)
+      }
+    })
+  })
+  ;[
+    [true, makeChannelPostList(), true],
+    [true, [], false],
+    [false, makeChannelPostList(), true],
+    [false, [], true]
+  ].forEach(([processing, posts, expectedLoaded]) => {
+    it(`sets loaded to ${String(expectedLoaded)} when there are ${
+      posts.length
+    } posts and API request processing=${String(processing)}`, async () => {
+      const { wrapper } = await render({
+        frontpage: {
+          processing: processing,
+          loaded:     !processing,
+          data:       {
+            postIds:    posts.map(post => post.id),
+            pagination: null
+          }
         }
-      )
+      })
+      assert.equal(wrapper.props().loaded, expectedLoaded)
+    })
+  })
+  ;[true, false].forEach(loaded => {
+    it(`sets canLoadMore=${String(loaded)} when loaded=${String(
+      loaded
+    )}`, async () => {
+      const { wrapper } = await render({
+        frontpage: {
+          processing: !loaded,
+          loaded:     loaded
+        }
+      })
+      assert.equal(wrapper.props().canLoadMore, loaded)
+    })
+  })
 
-      wrapper.update()
-      assert.equal(
-        wrapper.find(HomePage).props().location.search,
-        `?sort=${sortType}`
-      )
-    }
+  it("sets props used by withPostList", async () => {
+    const { wrapper } = await render()
+
+    assert.isFalse(wrapper.props().showPinUI)
+    assert.isTrue(wrapper.props().showReportPost)
+    assert.isFalse(wrapper.props().showRemovePost)
+    assert.isFalse(wrapper.props().showTogglePinPost)
+    assert.isTrue(wrapper.props().showChannelLinks)
+    assert.isFalse(wrapper.props().isModerator)
```

Comment:
You can set the result of `wrapper.props()` to a variable and access it instead of the repeated calls